rust/arrow/src/ffi.rs

```diff
@@ -117,20 +120,36 @@ unsafe extern "C" fn release_schema(schema: *mut FFI_ArrowSchema) {
     schema.release = None;
 }
```

Comment:
I'm surprised `release_schema` isn't updated. It should call the release callbacks of the children. Here is the C++ version:
https://github.com/apache/arrow/blob/master/cpp/src/arrow/c/bridge.cc#L101-L121